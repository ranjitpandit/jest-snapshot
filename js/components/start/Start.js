/**
 * @providesModule Start
 * @flow
 */
'use strict';

import React, {Component, PropTypes} from 'react'
import {StyleSheet, View, Text, TouchableHighlight} from 'react-native'

const styles = StyleSheet.create({

    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
    }
})

class MyComponent extends Component {

    constructor(props) {
        super(props)
        this.state = {}
    }

    render() {
        return (
            <View style={styles.container}>
                <TouchableHighlight
                                    onPress={this.onPressText.bind(this)}>
                    <Text>hi {this.props.name}</Text>
                </TouchableHighlight>
            </View>
        )
    }
    onPressText() {
        console.log('pressed')
        this.setState( {pressed: true})
    }
}


MyComponent.propTypes = {}

export default MyComponent;