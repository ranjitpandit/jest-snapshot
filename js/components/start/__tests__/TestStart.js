/**
 * Created by ranjit on 7/27/16.
 */
import 'react-native';
import React from 'react';
import Start from '../Start';

// Note: test renderer must be required after react-native.
import renderer from 'react/lib/ReactTestRenderer';

describe('Start', () => {

    it('renders correctly', () => {
        const output = renderer.create(
            <Start />
        )
        const tree = output.toJSON();
        expect(tree).toMatchSnapshot();

        // console.log(output)
        // console.dir(JSON.stringify(output))
        let instance = output.getInstance()
        output.getInstance().onPressText()
        // console.log('after onPress')
        // console.log(instance)

        expect(instance.state.pressed).toBe(true)
    });

});