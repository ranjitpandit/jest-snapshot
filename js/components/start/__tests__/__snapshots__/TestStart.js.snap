exports[`Start renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center"
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent"
        },
        undefined
      ]
    }
    testID={undefined}>
    <Text>
      hi 
    </Text>
  </View>
</View>
`;
